src/test/ui/associated-const/defaults-not-assumed-fail.rs

```diff
@@ -0,0 +1,44 @@
+// compile-fail
+
+trait Tr {
+    const A: u8 = 255;
+
+    // This should not be a constant evaluation error (overflow). The value of
+    // `Self::A` must not be assumed to hold inside the trait.
+    const B: u8 = Self::A + 1;
```

Comment:
So this wasn't exactly what I had in mind but it appears that testing that you cannot assume defaults inside a trait itself is not yet possible as lazy normalization isn't implemented. I've filed https://github.com/rust-lang/rust/issues/61815 for this issue to add a test later.